client/src/main/java/io/pravega/client/stream/impl/PositionImpl.java

```diff
@@ -147,7 +157,20 @@ private void write01(PositionImpl position, RevisionDataOutput revisionDataOutpu
             Map<Segment, Range> map = position.segmentRanges;
             revisionDataOutput.writeMap(map, (out, s) -> out.writeUTF(s.getScopedName()), PositionSerializer::writeRange);
         }
-        
+
+        private void write02(PositionImpl position, RevisionDataOutput revisionDataOutput) throws IOException {
+            // serialize offset segment again as CompactSignedLong ensure serialization
+            Map<Segment, Long> map = position.getOwnedSegmentsWithOffsets();
+            revisionDataOutput.writeMap(map, (out, s) -> out.writeUTF(s.getScopedName()),
+                    (out, offset) -> out.writeCompactSignedLong(offset));
```

Comment:
TIP: you can use a syntactic shortcut like `RevisionDataOutput::writeCompactSignedLong`